### PR TITLE
fix enconding to new new double/sint64 names

### PR DESCRIPTION
**Pull request by gordonguthrie**

### CODE REVIEW

**Comment by gordonguthrie** [comment]
Goes with these PRs
https://github.com/basho/riak_kv/pull/1247
https://github.com/basho/riak_ql/pull/36
https://github.com/basho/riak_test/pull/915

**Comment by alexmoore** [comment]
Works with the addition of https://github.com/basho/riak_pb/pull/153.

**Comment by matthewvon** [comment]
+1

**Comment by alexmoore** [comment]
Thanks for the review @matthewvon, going to stage it for bishop and merge when all the PRs are ready.

**Comment by alexmoore** [comment]
+1 f218ecbcc6a385834836f55ac21d8ab7d9bf650b

**Comment by DSomogyi** [comment]
RTS-509

File: src/eleveldb_ts.erl [hunk 1]
```diff
@@ -18,13 +18,11 @@ encode_key(Elements) when is_list(Elements) ->
...
-encode_k2([{integer, I}    | T], Bin) -> encode_k2(T, append(<<I:64/integer>>, Bin));
-encode_k2([{binary, B}     | T], Bin) when is_binary(B) -> 
-    encode_k2(T, append(B, Bin));
-encode_k2([{binary, L}     | T], Bin) when is_list(L) ->
-    B = list_to_binary(L),
-    encode_k2(T, append(B, Bin)).
+encode_k2([{double, F}     | T], Bin) -> encode_k2(T, append(<<F:64/float>>,  Bin));
+encode_k2([{sint64, I}     | T], Bin) -> encode_k2(T, append(<<I:64/integer>>, Bin));
+encode_k2([{varchar, B}    | T], Bin) when is_binary(B) -> encode_k2(T, append(B, Bin));
+encode_k2([{binary, L}     | T], Bin) when is_list(L)   -> B = list_to_binary(L),
```
Comment (andytill) [inline src/eleveldb_ts.erl]: Looking at the tests in riak_ql, a binary type is always a binary and not a list.

https://github.com/basho/riak_ql/blob/develop/src/riak_ql.yrl.tests#L257

Comment (andytill) [inline src/eleveldb_ts.erl]: The atom `binary` gets changed to the atom `varchar` in `riak_kv_qry_compiler:rewrite/3` so the binary clause here is dead code.

Comment (alexmoore) [inline src/eleveldb_ts.erl]: Can we safely delete that then?

Comment (andytill) [inline src/eleveldb_ts.erl]: Yes, the riak_test suites use varchars for puts and selects so this should be well covered by tests to prove it.

**Comment by lukebakken** [comment]
Output of `make test`:

``` sh
======================== EUnit ========================
module 'eleveldb_ts'
  eleveldb_ts: simple_encode_key_test...ok
  eleveldb_ts: simple_encode_record_test...[0.006 s] ok
  eleveldb_ts: simple_decode_record_test...ok
  [done in 0.015 s]
module 'eleveldb_bump'
  [done in 0.001 s]
module 'basho_bench_driver_eldb'
module 'iterators'
enif_send: env==NULL on non-SMP VMtools.mk:7: recipe for target 'test' failed
make: *** [test] Aborted (core dumped)
```

**Comment by lukebakken** [comment]
+1 0bf857ee9f7a1a01bafc83c48391b63b6d1cbb87

**Comment by lukebakken** [comment]
@borshop retry

**Comment by lukebakken** [comment]
@borshop merge

**Pull request merged**